server/middleware/request-validation-middlewares.ts

```diff
@@ -0,0 +1,18 @@
+import { Request, Response, NextFunction } from "express";
+import Ajv from 'ajv';
+const ajv = new Ajv({ allErrors: true });
+
+export const requestValidationMiddleware = (schema: any) => {
+  
+  const validatingRequest = ajv.compile(schema);
+
+  return (req: Request, res: Response, next: NextFunction) => {
+    const valid = validatingRequest(req.body);
+    if (!valid) {
+      console.log(validatingRequest.errors);
+      res.sendStatus(400);
+    }
+    else
+      next();
+  }
+}
```

Comment:
instead of using `any`, I think we can do
```
export function requestValidationMiddleware<T>(schema: T) {
  const validatingRequest = ajv.compile(schema as object);

  return (req: Request, res: Response, next: NextFunction) => {
    const valid = validatingRequest(req.body);
    if (!valid) {
      console.log(validatingRequest.errors);
      res.sendStatus(400);
    } else next();
  };
}
```

Reply:
Thank you, this was very helpful